registry/reference.go

```diff
@@ -54,8 +54,19 @@ type Reference struct {
 	// Reference is the reference of the object in the repository.
 	// A reference can be a tag or a digest.
 	Reference string
+
+	ReferenceType ReferenceType
```

Comment:
We should not introduce `ReferenceType` since the sets of tags and digests do not have intersection. Introducing `ReferenceType` will make the design unnecessarily overcomplex.

Reply:
Taking a different approach in order to avoid modification of this struct.